src/transports/ipc.rs

```diff
@@ -0,0 +1,185 @@
+                Some(Ok(bytes)) => {
+                    read_buffer.extend_from_slice(&bytes);
+
+                    let pos = match read_buffer.iter().rposition(|&b| b == b']' || b == b'}') {
+                        Some(pos) => pos,
+                        None => continue,
+                    };
+
+                    match helpers::to_response_from_slice(&read_buffer[..pos]) {
+                        Ok(response) => {
+                            let remaining_bytes_len = read_buffer.len() - pos;
+                            for i in 0..remaining_bytes_len {
+                                read_buffer[i] = read_buffer[pos + i];
+                            }
+                            read_buffer.truncate(remaining_bytes_len);
+
+                            let outputs = match response {
+                                rpc::Response::Single(output) => vec![output],
+                                rpc::Response::Batch(outputs) => outputs,
+                            };
+
+                            for output in outputs {
+                                let id = match &output {
+                                    rpc::Output::Success(success) => success.id.clone(),
+                                    rpc::Output::Failure(failure) => failure.id.clone(),
+                                };
+
+                                let value = match helpers::to_result_from_output(output) {
+                                    Ok(value) => value,
+                                    Err(err) => {
+                                        log::warn!("Unable to parse output into rpc::Value: {:?}", err);
+                                        continue;
+                                    },
+                                };
+
+                                let id = match id {
+                                    rpc::Id::Num(num) => num as usize,
+                                    _ => {
+                                        log::warn!("Got unsupported response (id: {:?})", id);
+                                        continue;
+                                    },
+                                };
+
+                                let response_tx = match pending_response_txs.remove(&id) {
+                                    Some(response_tx) => response_tx,
+                                    None => {
+                                        log::warn!("Got response for unknown request (id: {:?})", id);
+                                        continue;
+                                    },
+                                };
+
+                                if let Err(err) = response_tx.send(value) {
+                                    log::warn!("Sending a response to deallocated channel: {:?}", err);
+                                }
+                            }
+                        }
+                        Err(err) => log::warn!("Got bad IPC response bytes: {:?}", err),
```

Comment:
Should it really be a `warn`? What if we simply didn't get a full response yet (but it does have `}` or `]`)?

Reply:
Nice catch

Reply:
Done in 6a28cade6a92ba553f77afdfac95860a4873b5ca